### PR TITLE
#124 Allow processorTimeout property to be configured by API

### PR DESCRIPTION
**Pull request by mcweba**

### CODE REVIEW

File: src/main/java/org/swisspush/redisques/util/DefaultRedisquesConfigurationProvider.java [hunk 1]
```diff
@@ -46,19 +48,18 @@ public Result<Void, String> updateConfiguration(JsonObject configuration, boolea
...
+        if (configurationValues != null && !configurationValues.isEmpty()) {
             List<String> notAllowedConfigurationValues = findNotAllowedConfigurationValues(configurationValues.fieldNames());
             if (notAllowedConfigurationValues.isEmpty()) {
                 try {
                     Long processorDelayMaxValue = configurationValues.getLong(PROCESSOR_DELAY_MAX);
-                    if (processorDelayMaxValue == null) {
-                        return Result.err("Value for configuration property '" + PROCESSOR_DELAY_MAX + "' is missing");
-                    }
+                    Long processorTimeoutValue = configurationValues.getLong(PROCESSOR_TIMEOUT);
+
```
Comment (hiddenalpha) [inline src/main/java/org/swisspush/redisques/util/DefaultRedisquesConfigurationProvider.java:57]:
```patch
-  if (processorDelayMaxValue == null) {
-      return Result.err("Value for configuration property '" + PROCESSOR_DELAY_MAX + "' is missing");
-  }
```
I assume it is by intent to remove this check?

File: src/main/java/org/swisspush/redisques/util/RedisquesConfiguration.java [hunk 2]
```diff
@@ -42,6 +42,7 @@ public class RedisquesConfiguration {
     private final int memoryUsageLimitPercent;
     private final int memoryUsageCheckIntervalSec;
     private static final int DEFAULT_CHECK_INTERVAL = 60; // 60s
+    private static final int DEFAULT_PROCESSOR_TIMEOUT = 240000; // 240S
```
Comment (hiddenalpha) [inline src/main/java/org/swisspush/redisques/util/RedisquesConfiguration.java:45]: Suggestion:
```patch
-   240000; // 240S
+  240_000;
```
If not, then I suggest to at least use a *small* s for seconds ;)

Comment (hiddenalpha) [inline src/main/java/org/swisspush/redisques/util/RedisquesConfiguration.java:45]: (*trivial*) Suggest:
```patch
-  ... DEFAULT_PROCESSOR_TIMEOUT ...
+  ... DEFAULT_PROCESSOR_TIMEOUT_MS ...
```
IMHO in such circumstances it is a good idea to make the unit more clear.

**Review by stettlermari — APPROVED** [review]

File: src/main/java/org/swisspush/redisques/util/DefaultRedisquesConfigurationProvider.java (hunk 1 above)
Comment (mcweba) [reply src/main/java/org/swisspush/redisques/util/DefaultRedisquesConfigurationProvider.java:57]: Yes it is by intent

**Pull request merged**